### DISCUSSION TITLE
Getting the client/server example to run with pyright

### DISCUSSION THREAD

**Discussion by philllies** [TypeFox/monaco-languageclient#510] [Q&A]
I'm trying to get the example to run with pyright, using the following line in `json-server-launcher.ts`:

```
const serverConnection = createServerProcess('JSON', 'pyright-langserver', ['--stdio']);
```

The initialization/getting the capabilities part seems to work fine, but then it stalls on any subsequent completion requests.

Should this just work out of the box? It seems so because it does work for `pylsp`.

**Comment by msujew** [comment]
I did some work with pyright in the recent months in combination with monaco. It was for a web compatible version of pyright though, see [here](https://www.npmjs.com/package/@typefox/pyright-browser). Maybe that works for you as well.

I'm not sure what's wrong with the node based version though. Maybe it failed to start the analysis worker thread? See [here](https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/nodeServer.ts).

**Comment by kaisalmen** [comment]
We opened issue #521

**Comment by kaisalmen** [comment]
We want to integrate a pyright example directly here.
